simpleclient_servlet/pom.xml

```diff
@@ -84,5 +84,5 @@
             <scope>test</scope>
         </dependency>
     </dependencies>
-
+    <!--This cannot have an automatic module name as its packages conflicts with the simpleclient module-->
```

Comment:
I guess this is directly related to my question here: https://github.com/prometheus/client_java/issues/607#issuecomment-952310842.

As I have never used Jigsaw, could you please explain why we cannot expose the packages `io.prometheus.client.exporter` and `io.prometheus.client.filter` here so that users can use the `MetricsServlet` and the `MetricsFilter`? Sorry if this is a stupid question, I have no experience with Java modules.

Reply:
Yeah, so it's a bit of a backwards compatibility nightmare. There are two kinds of modules: one with `module-info.class` and automatic ones.

If you create your own `module-info` then you can control what you export, but you can't create classes in any packages from modules that you import, so your `Adapter` class would fail to compile.

For automatic modules, because they're backwards compatible, you have to export all the classes you have and only one module can export a package.

Oracle claimed it was always best practice so have each package provided by only one JAR, but that was never taken seriously. Now with the new module system, it's law. The logic is that Java's "friend" permission-level meant that only classes in the same package could access a certain class or method or field or whatever. Since you could always just create a class in the same package, it had no real security enforcement. This restriction is meant to make friend permissions be useful.

Reply:
Upon closer inspection: 
`simpleclient_httpserver` and `simpleclient_pushgateway` both have classes in the `io.prometheus.client.exporter`. That means no module can import both of them at the same time. I can't imagine a use case where both are necessary, so it can be ignored for now, but, if these packages are going to be refactored in the future, they should be renamed.